classes/04class/exercises/c04-iac02/gpavelar/vpc.tf

```diff
@@ -0,0 +1,119 @@
+##
+# VPC
+#
+resource "aws_vpc" "devopsacademy-iac" {
+  cidr_block           = var.vpc_cidr
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  tags = {
+    Name = var.vpc_name
+  }
+}
+
+##
+# Internet Gateway
+#
+resource "aws_internet_gateway" "c04-iac-igw" {
+  vpc_id = aws_vpc.devopsacademy-iac.id
+
+  tags = {
+    Name = var.igw-name
+  }
+}
+
+##
+# Subnets
+#
+
+resource "aws_subnet" "private-subnet" {
+  count             = length(var.subnets_cidrs_private)
+  vpc_id            = aws_vpc.devopsacademy-iac.id
+  cidr_block        = var.subnets_cidrs_private[count.index]
+  availability_zone = var.azs[count.index]
+
+  tags = {
+    Name = "${var.subnets_names_private[count.index]}"
+  }
+
+  depends_on = [aws_vpc.devopsacademy-iac]
+}
+
+resource "aws_subnet" "public-subnet" {
+  count                   = length(var.subnets_cidrs_public)
+  vpc_id                  = aws_vpc.devopsacademy-iac.id
+  cidr_block              = var.subnets_cidrs_public[count.index]
+  map_public_ip_on_launch = true
+  availability_zone       = var.azs[count.index]
+
+  tags = {
+    Name = "${var.subnets_names_public[count.index]}"
+  }
+
+  depends_on = [aws_vpc.devopsacademy-iac]
+}
+
+##
+# NAT Gateway
+#
+resource "aws_eip" "eip-nat" {
+  vpc = true
+}
+
+resource "aws_nat_gateway" "nat-1" {
+  count = 1
```

Comment:
you don't need to use count feature here since it will always have one nat_gateway.

It would be fine if you like to use some `feature_flag` feature where you can enable/disable the deployment of the resource based on variables, but it is not necessary here.